### PR TITLE
Ensure we can capture non-built-in modules with 'require'.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/pulumi/pulumi/issues/1676

Previous work only ensured that built-in modules captured properly.  This expands this out to any module we find in the node_modules path.

### CODE REVIEW

File: sdk/nodejs/runtime/closure/createClosure.ts [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ function getOrCreateEntry(
     }
 
     function captureModule(moduleName: string) {
-        if (moduleName[0] === ".") {
+        if (moduleName.startsWith(".") && !moduleName.startsWith("./node_modules/")) {
```
Comment (lukehoban) [inline sdk/nodejs/runtime/closure/createClosure.ts]: Just for future sanity - we may want to add a comment to `findModuleName` with examples of the paths it returns.  The paths stored in the Node module cache are not what you might naively expect, so interpreting these correctly is easy to get wrong.

Comment (lukehoban) [inline sdk/nodejs/runtime/closure/createClosure.ts]: Can we simplify this to just checking for `moduleName.startsWith("./node_modules/")` and doing what is currently the `else` branch below, and leave the "else" branch to cover the "." case?

Also - the `else if (moduleName.startsWith("@pulumi")) {` is probably wrong too - right?  I assume this comes back from `findModuleName` as `./node_modules/@pulumi/pulumi`?

**Comment by lukehoban** [comment]
As part of this change, would be good to update https://github.com/pulumi/pulumi-aws/blob/master/tests/serverless_functions/index.ts to take advantage of this and do more complete end-to-end validation across all of the now supported scenarios.

File: sdk/nodejs/runtime/closure/createClosure.ts (hunk 1 above)
Comment (CyrusNajmabadi) [reply sdk/nodejs/runtime/closure/createClosure.ts]: right.  the @pulumi part is likely wrong.  i'll use this as the opportunity to add the "deploymentOnly" module bit.

**Comment by CyrusNajmabadi** [comment]
> As part of this change, would be good to update https://github.com/pulumi/pulumi-aws/blob/master/tests/serverless_functions/index.ts

perfect, thanks.  i was looking for the best way to validate this.

File: sdk/nodejs/runtime/closure/createClosure.ts (hunk 1 above)
Comment (CyrusNajmabadi) [reply sdk/nodejs/runtime/closure/createClosure.ts]: sounds good.

Comment (CyrusNajmabadi) [reply sdk/nodejs/runtime/closure/createClosure.ts]: Done.

Comment (CyrusNajmabadi) [reply sdk/nodejs/runtime/closure/createClosure.ts]: Done.

**Comment by CyrusNajmabadi** [comment]
@lukehoban Ready for another review.

**Comment by CyrusNajmabadi** [comment]
I'm validating that this works in a variety of differnet configurations.

**Comment by CyrusNajmabadi** [comment]
Have validated with multiple interesting packages, and multiple local module arrangements.  Everything seems to work well here.  Additional tests forthcoming in https://github.com/pulumi/pulumi-aws/pull/287 once this goes in.

**Pull request merged**